### DISCUSSION TITLE
Question: freeplaystate.hx

### DISCUSSION THREAD

**Discussion by Toppat-dalek** [FunkinCrew/Funkin#1521] [Q&A] [answered]
## What is your question?
did you change freeplaystate.hx state to automatically add all songs? because I cant seem to find where the names of the songs are in freeplaystate.hx.

**Comment by Two-Ai** [comment, marked as answer]
`FreeplayState.hx` loads the list of songs from `freeplaySonglist.txt`, then adds the story songs manually.
https://github.com/ninjamuffin99/Funkin/blob/master/source/FreeplayState.hx#L38
https://github.com/ninjamuffin99/Funkin/blob/master/source/FreeplayState.hx#L64-L80

**Comment by aflacc** [comment]
its a little complicated but you can add songs directly thru the txt but with the gf icon only, or you can add them with `freeplaystate.hx` and have an icon. this way also lets you set the order
